SimCalorimetry/EcalSimProducers/src/EcalDigiProducer_Ph2.cc

```diff
@@ -0,0 +1,375 @@
+      m_PreMix2(params.getParameter<bool>("EcalPreMixStage2")),
+
+      m_doEB(params.getParameter<bool>("doEB")),
+
+      m_APDDigitizer(nullptr),
+      m_BarrelDigitizer(nullptr),
+      m_ElectronicsSim(nullptr),
+      m_Coder(nullptr),
+      m_APDElectronicsSim(nullptr),
+      m_APDCoder(nullptr),
+      m_Geometry(nullptr),
+      m_EBCorrNoise({{nullptr, nullptr}})
+
+{
+  //std::cout<<"[EcalDigiProducer_Ph2] Constructing producer"<<     "frameSize: " <<  m_readoutFrameSize<< std::endl;
+  if (m_doEB)
+    iC.consumes<std::vector<PCaloHit> >(edm::InputTag(m_hitsProducerTag, "EcalHitsEB"));
+
+  const std::vector<double> ebCorMatG10Ph2 = params.getParameter<std::vector<double> >("EBCorrNoiseMatrixG10Ph2");
+  const std::vector<double> ebCorMatG01Ph2 = params.getParameter<std::vector<double> >("EBCorrNoiseMatrixG01Ph2");
+
+  const bool applyConstantTerm = params.getParameter<bool>("applyConstantTerm");
+  const double rmsConstantTerm = params.getParameter<double>("ConstantTerm");
+
+  const bool addNoise = params.getParameter<bool>("doENoise");
+  const bool cosmicsPhase = params.getParameter<bool>("cosmicsPhase");
+  const double cosmicsShift = params.getParameter<double>("cosmicsShift");
+
+  //++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  // further phase for cosmics studies
+  if (cosmicsPhase) {
+    if (m_doEB)
+      m_EBResponse->setPhaseShift(1. + cosmicsShift);
+  }
+
+  EcalCorrMatrix_Ph2 ebMatrix[2];
+
+  //std::cout<<"[EcalDigiProducer_Ph2] Check size10: "<<ebCorMatG10Ph2.size()<<std::endl;
+  //std::cout<<"[EcalDigiProducer_Ph2] Check size01: "<<ebCorMatG01Ph2.size()<<std::endl;
+  assert(ebCorMatG10Ph2.size() == m_readoutFrameSize);
+  assert(ebCorMatG01Ph2.size() == m_readoutFrameSize);
+
+  //std::cout<<"[EcalDigiProducer_Ph2] Check Cor Mat consistency"<<std::endl;
+  assert(1.e-7 > fabs(ebCorMatG10Ph2[0] - 1.0));
+  assert(1.e-7 > fabs(ebCorMatG01Ph2[0] - 1.0));
+
+  //std::cout<<"[EcalDigiProducer_Ph2] Check row of cor and col mat"<<std::endl;
+  for (unsigned int row(0); row != m_readoutFrameSize; ++row) {
+    assert(0 == row || 1. >= ebCorMatG10Ph2[row]);
+    assert(0 == row || 1. >= ebCorMatG01Ph2[row]);
+
+    // std::cout<<"row "<<row<<std::endl;
+
+    for (unsigned int column(0); column <= row; ++column) {
+      const unsigned int index(row - column);
+      // std::cout<<"index: "<<index<<std::endl;
+      // std::cout<<"ebCorMat10: "<<ebCorMatG10Ph2[ index ] <<std::endl;
+      // std::cout<<"ebCorMat01: "<<ebCorMatG01Ph2[ index ] <<std::endl;
+      ebMatrix[0](row, column) = ebCorMatG10Ph2[index];
+      //std::cout<<"col mat0: "<<column<<std::endl;
+      ebMatrix[1](row, column) = ebCorMatG01Ph2[index];
+      //std::cout<<"col mat1 "<<column<<std::endl;
+    }
+  }
+  //std::cout<<"[EcalDigiProducer_Ph2] Reset"<<std::endl;
+  m_EBCorrNoise[0].reset(new CorrelatedNoisifier<EcalCorrMatrix_Ph2>(ebMatrix[0]));
```

Comment:
use `std::make_unique` rather than `new `
(in every line below)

Reply:
`m_EBCorrNoise[0] = std::make_unique<CorrelatedNoisifier<EcalCorrMatrix_Ph2>>(ebMatrix[0]);`

Reply:
Thanks, I modify all of them like this, i didn't know how to do it